pandas/_libs/hashtable.pyx

```diff
@@ -142,7 +142,7 @@ cdef class Int64Factorizer(Factorizer):
         --------
         Factorize values with nans replaced by na_sentinel
 
-        >>> factorize(np.array([1,2,np.nan], dtype='O'), na_sentinel=20)
+        >>> factorize(np.array([1,2,np.nan], dtype='O'), na_sentinel=20)# doctest: +SKIP
```

Comment:
why does this one need skipping?

Reply:
not sure why, it's not able to find `factorize` definition (also had similar issues in timezones.pyx)
would you have any idea?
```
_______________________________________________________________________ [doctest] pandas._libs.hashtable.Int64Factorizer.factorize _______________________________________________________________________
EXAMPLE LOCATION UNKNOWN, not showing all tests of that example
??? >>> factorize(np.array([1,2,np.nan], dtype='O'), na_sentinel=20)
UNEXPECTED EXCEPTION: NameError("name 'factorize' is not defined")
Traceback (most recent call last):
  File "/Users/debnathshoham/miniconda3/envs/pandas-dev/lib/python3.8/doctest.py", line 1336, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest pandas._libs.hashtable.Int64Factorizer.factorize[0]>", line 1, in <module>
NameError: name 'factorize' is not defined
/Volumes/SD/pandas-debnathshoham/pandas/_libs/hashtable.cpython-38-darwin.so:None: UnexpectedException
______________________________________________________________________ [doctest] pandas._libs.hashtable.ObjectFactorizer.factorize _______________________________________________________________________
EXAMPLE LOCATION UNKNOWN, not showing all tests of that example
??? >>> factorize(np.array([1,2,np.nan], dtype='O'), na_sentinel=20)
UNEXPECTED EXCEPTION: NameError("name 'factorize' is not defined")
Traceback (most recent call last):
  File "/Users/debnathshoham/miniconda3/envs/pandas-dev/lib/python3.8/doctest.py", line 1336, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest pandas._libs.hashtable.ObjectFactorizer.factorize[0]>", line 1, in <module>
NameError: name 'factorize' is not defined
/Volumes/SD/pandas-debnathshoham/pandas/_libs/hashtable.cpython-38-darwin.so:None: UnexpectedException

Reply:
I don't know, perhaps you need the class name? (I haven't tried this)

Reply:
Thanks! that did the trick in `ObjectFactorizer`.

Although for `IntFactorizer`, I am struggling to understand the use-case. The array can't contain `NaN` and be `int64`, am I missing something? (the original docstring was `object` dtype, which I think was incorrect)

Reply:
I don't know, I've never used it - perhaps check `git log` and see if you can someone who's worked on it?

Reply:
@jbrockmendel I see you have made some changes in `IntFactorizer`. Could you help me understand the use-case?

> The array can't contain NaN and be int64, am I missing something? (the original docstring was object dtype, which I think was incorrect)